project-management/src/main/java/life/qbic/projectmanagement/application/measurement/MeasurementService.java

```diff
@@ -414,10 +312,55 @@ private Map<ProteomicsMeasurement, Collection<SampleIdCodeEntry>> mergeByPool(
     return metadataMap;
   }
 
+  private Map<NGSMeasurement, Collection<SampleIdCodeEntry>> mergeByPoolNGS(
+      Map<String, List<NGSMeasurementMetadata>> groupedMetadata, ProjectId projectId) {
+    Map<NGSMeasurement, Collection<SampleIdCodeEntry>> metadataMap = new HashMap<>();
+    for (var metadataGroup : groupedMetadata.entrySet()) {
+      metadataMap.putAll(buildNGS(metadataGroup.getValue(), projectId));
+    }
+    return metadataMap;
+  }
+
+  private Map<NGSMeasurement, Collection<SampleIdCodeEntry>> buildNGS(
```

Comment:
Please rename the proteomics method accordingly. `buildPxp` instead of `build`.

As you are using the same method structure except for the type of grouped metadata and the call to a different build method, maybe there is some abstraction waiting here? For now keeping it separate makes sense :+1:

Reply:
I heavily agree that we need to abstract the measurement itself instead of having duplicated code everywhere. 👍